src/core/ts/background-scripts/background.js/telemetry/Telemetry.ts

```diff
@@ -117,17 +135,15 @@ export class Telemetry {
 
   public onTranslateButtonPressed(tabId: number, from: string, to: string) {
     this.submit(() => {
```

Comment:
We shouldn't submit ping for all metrics. Instead, we should aim at submitting one ping per user-page interaction scenario.
1. Regular translation: infobar displayed -> translated pressed -> translation finished or error -> submit
2. Switch language: infobar displayed -> switch source or target lang -> submit (further actions will be submitted in translation scenario)
3. Reject: infobar displayed -> press never, now now or close -> submit
4. No action: infobar displayed -> no action -> submit on timer after some time (might be harder to implement though).
5. Language pair unsupported -> submit

I think scenario 2 will not happen often and then we'll have 1 ping per page in most cases.
As for service metrics, mismatch with supported lang pair will be submitted together with other scenarios and when unsupported we can submit it on its own.

We had a deal that it will be fixed in this PR as far as I remember :) But let's discuss if you have any objections about all this.

Reply:
The deal previously was to fix it in Milestone 3, but got bumped to Milestone 4 when we spoke this week, at least so I understood it. Happy to move it back to Milestone 3 but if ok with you I'd love to land incremental PRs, so please approve this PR if everything else is satisfactory and I promise to get a new PR up soon. It will also make review of the ping grouping a bit easier if there is a dedicated PR for it.

Reply:
ok, but it would be great to land it before integration in Nightly